Add build and deploy templates for openshift operations

carbonin opened the pull request:
~~This is not added to the deployment script as the intention
is for this to be deployed by the orchestrator.~~

We are going to deploy this unconditionally and maybe do some dynamic scaling in the future. Not having any openshift source at all is too unrealistic a condition to add a who new section to the orchestrator.

This also renames the image for the openshift collector and shares it across the two deployments. The operations worker overrides the entrypoint.

@agrare @eclarizio Let me know what you think about the environment variables and if you think we need anything else.

agrare commented:
:+1: LGTM

eclarizio reviewed (approved):
👍 I don't see anything else I think we need at the moment, they look fine to me.

eclarizio commented:
Actually, @carbonin, with the topological-inventory-api-client, we will need a username/password for basic auth. Do we think this should be included here or somewhere else?

carbonin committed:
Rename the collector image to topological-inventory-openshift

The collector and operations workers now share a repo and they can
also share an image.
The entrypoint by default will be the collector bin script so that
we don't have to encode the entrypoint for each collector into the
orchestrator, but we can override that in the operations worker
deployment config.

carbonin committed:
Add the openshift operations deployment template

carbonin commented:
Removing WIP as https://github.com/ManageIQ/topological_inventory-api/pull/103 looks about ready to go.

@bdunne @agrare want to take another look?

agrare reviewed (approved):
:+1: LGTM

eclarizio reviewed (approved):
👍
